controllers/goals.js

```diff
@@ -0,0 +1,44 @@
+const config = require("config");
+const { SOMETHING_WENT_WRONG } = require("../constants/errorMessages");
+
+const getGoalSiteToken = async (req, res) => {
+  try {
+    const { roles, id: userId } = req.userData;
+    const goalSiteConfig = config.services.goalAPI;
+
+    const goalApiResponse = await fetch(`${goalSiteConfig.baseUrl}/auth_token`, {
+      method: "POST",
+      body: JSON.stringify({
+        user_id: userId,
+        roles,
+        goal_api_secret_key: goalSiteConfig.secretKey,
+      }),
```

Comment:
Why are we using stringify here?

Reply:
Using the browser fetch api, there we have to send json, right?